### DISCUSSION TITLE
Should the catalog allow "non-silent" installers?

### DISCUSSION THREAD

**Discussion by denelon** [microsoft/winget-pkgs#9190] [Q&A]
The current logic in validation, excludes installers that require user interaction. One of the main challenges is the complexity of verifying these packages install. It would likely require additional manual validation. The other challenge is that it would prevent users from being able to install multiple packages with import, and being able to "walk away" and come back to a completed install.

If we were to begin allowing interactive installers, we would likely also need to add extra meta-data for search to identify these kinds of packages, and optionally ignore or avoid them for users who are looking for silent/scriptable installs.

The issue below represents and example of a package that once supported silent install, and now appears not to.
https://github.com/microsoft/winget-pkgs/issues/9185

**Comment by auanasgheps** [comment]
I think the catalog should allow for non-silent installers, with a clear note so the user will know some interaction is needed.

There are some _almost_ silent installers that require an additional confirmation, like a driver installation, which has to be there.

Another example is Logitech software, most of that is not silent anymore (and we should blame the developer for this), but the installation is just one click away.

Some rules could be defined, eg: no more than X clicks, no bundled PUA, and so on.
Or you could simply evaluate approval on a case by case scenario.

**Reply by denelon** [reply]
I've seen several requests to support remote installation via SSH or similar, and those likely wouldn't work at all with any interaction. How about a setting or filter to "hide" interactive packages from users who want only silent?

**Reply by auanasgheps** [reply]
Oh yes, a filter/setting sounds good!

**Reply by denelon** [reply]
https://github.com/microsoft/winget-cli/issues/823 needs to be implemented, and then we can look at relaxing the restriction on the community repository "Open Windows Catalog".
